### PR TITLE
Feature/3075/tools version table

### PR DESCRIPTION
**Pull request by garyluu**
For remaining part of dockstore/dockstore#3075

Converting tools versions table to Material.  

Also added JAVA_TOOL_OPTIONS to match the webservice's CircleCI

Still missing a bit of styling, but that's in another issue.

image

### CODE REVIEW

**Comment by garyluu** [comment]
Will fix prettier merge conflicts after review changes

File: src/app/container/versions/versions.component.html [hunk 1]
```diff
@@ -15,153 +15,124 @@
...
+        <div *ngIf="tool?.mode !== DockstoreToolType.ModeEnum.HOSTED">
+          <i *ngIf="version.referenceType === 'BRANCH'" class="fa fa-code-fork" aria-hidden="true" title="Branch"></i>
+          <i *ngIf="version.referenceType === 'TAG'" class="fa fa-tag" aria-hidden="true" title="Tag"></i>
+          <span
+            *ngIf="version.commitID && (version.commitID | commitUrl: tool?.providerUrl); else noCommitID"
+            [matTooltip]="version.commitID ? 'Commit ID: ' + version.commitID : ''"
+          >
+            &nbsp;<a [href]="version.commitID | commitUrl: tool?.providerUrl">{{ version.reference || 'n/a' }}</a>
+          </span>
+          <ng-template #noCommitID
```
[the remark this answers is not in the document]
Comment (garyluu) [reply src/app/container/versions/versions.component.html]: Just didn't use a template, not worth the effort.  This is was here before.  Still no idea if it affects material tables

File: src/app/container/container.component.html [hunk 2]
```diff
@@ -347,3 +349,16 @@ <h3>DOI</h3>
     </div>
   </div>
 </div>
+<ng-template #noVersions>
+  <div class="p-3">
+    <mat-card class="alert alert-info p-3" role="alert">
+      <mat-icon>info</mat-icon>
+      <span *ngIf="tool?.mode === ModeEnum.HOSTED">
+        To see versions, please add a new version.
```
[the remark this answers is not in the document]
Comment (garyluu) [reply src/app/container/container.component.html]: This is actually copied from the workflow counterpart to remain consistent.  It was already outside the scope of the PR.

[the remark this answers is not in the document]
Comment (denis-yuen) [reply src/app/container/container.component.html]: I think a whole new ticket is overkill, I really had in mind changing the comment to "To see versions, please add a new version from the [Edit FIles tab](link here)" kind of thing.

File: src/app/container/versions/versions.component.html [hunk 3]
```diff
@@ -15,153 +15,124 @@
...
+      <td mat-cell *matCellDef="let version">
+        <mat-icon *ngIf="version.hidden">check</mat-icon>
+      </td>
+    </ng-container>
+    <ng-container matColumnDef="verified">
+      <th
+        scope="col"
+        mat-header-cell
+        *matHeaderCellDef
+        matTooltip="A version is verified if it has been verified to work by a person/group."
```
Comment (denis-yuen) [inline src/app/container/versions/versions.component.html]: Maybe `A version is verified if it has been verified to work by a third party`

File: src/app/container/container.component.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (garyluu) [reply src/app/container/container.component.html]: A few things:
1. As you've said before, issues are free and so are PRs.
2. This is might be the perfect example of why it should be a new issue.  We don't actually have a tab specific link in my-tools/my-workflows.  If I were to modify this alert to your suggestion, you can expect the # of lines changed to double.
3. Knowing the efforts it entails, we should discuss as a group whether this is the solution we want

File: src/app/container/versions/versions.component.html (hunk 3 above)
Comment (garyluu) [reply src/app/container/versions/versions.component.html]: This was carried over from before. It matches the workflow counterpart. I don't mind changing both of them right now, but it seems the suggestion is kind of inaccurate given that it's mostly me verifying them.

Comment (denis-yuen) [reply src/app/container/versions/versions.component.html]: I'm ok with it being aspirational. I think the main thing is just to convey the idea that it isn't the author themselves verifying it.  Whether we use the term "third party", "outside party" as here https://docs.dockstore.org/en/develop/advanced-topics/verification.html , I'm open to ideas. 

But currently it seems unclear to me.

File: src/app/container/container.component.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (denis-yuen) [reply src/app/container/container.component.html]: *shrug* created an issue https://github.com/dockstore/dockstore/issues/3193
let's see some mock-ups and we can work from there

File: src/app/container/versions/versions.component.html (hunk 3 above)
Comment (garyluu) [reply src/app/container/versions/versions.component.html]: I changed the tooltip to what Solomon said.  Basically "if the column says verified platform, then the tooltip should also say it"

**Pull request merged**